src/python/pants/backend/jvm/tasks/jvm_compile/rsc/rsc_compile.py

```diff
@@ -723,9 +721,12 @@ def _runtool(self, main, tool_name, args, distribution, tgt=None, input_files=tu
         cmd.extend([main])
         cmd.extend(args)
 
+        input_digest = self.context._scheduler.capture_snapshots((root,))[0].directory_digest
```

Comment:
It could be handy to avoid some no-op work here in some situations; in particular:
* If `pathglobs` is empty, we shouldn't bother capturing a snapshot for it
* If `input_snapshot` is `None`, or `pathglobs` was empty, we shouldn't bother merging directories
(I haven't tested, but my assumption is that calling synchronously into the engine to ask it to capture or merge no-op things is non-trivial, albeit hopefully still cheap)

Reply:
> I haven't tested, but my assumption is that calling synchronously into the engine to ask it to capture or merge no-op things is non-trivial, albeit hopefully still cheap

If I had to guess, I'd say it was on the order of tens of milliseconds... possibly not worth the special case.